pkg/transforms/policyreport.go

```diff
@@ -117,8 +124,11 @@ func PolicyReportResourceBuilder(pr *PolicyReport) *PolicyReportResource {
 	policyList := policies.UnsortedList()
 	sort.Strings(policyList)
 
-	// "rules" is incorrect since there is a "rule" field in the results, but this is kept for backwards compatibility
-	node.Properties["rules"] = policyList
```

Comment:
This change is equivalent to an API change, so we need to be careful because if a client is using data from `rules` this will cause a break. We haven't made any change like this before, so we'll need to figure out how to handle it.

PolicyReport is used from the console code, but I'm not sure if we use the `rules` field.  The bigger risk is that customers could be using this data.

Reply:
FYI @zlayne 
Do you use the `rules` field from PolicyReport in the console?

Reply:
Not the best example becasue there aren't any violations here, but yes we show the rules in the PolicyReport search result table.
![Screenshot 2024-12-19 at 8 18 43 AM](https://github.com/user-attachments/assets/6f1aacfe-bb14-43e8-a7e1-e13c98d8f615)

Reply:
How about change to "ruleNames"

Reply:
@yiraeChristineKim Yes, `ruleNames` works, adding a new field won't break any client.

Now I'm also wondering, do we show **incorrect** data in the screenshot that @zlayne added above? If that's the case we can see his as a bug instead of a name change. If the API is not working as intended, then it's easier to justify the breaking change. It's subjective and has a risk, but it's better to return the correct data.

Reply:
I check on the console. The rule columns represent policy names which are applied clusterpolicy, policy.  For example `apiVersion: wgpolicyk8s.io/v1beta1
kind: PolicyReport
metadata:
  name: 3d4d8287-57bf-4d79-a5ff-a7cc0774e11b
  namespace: test1
results:
- message: 'validation error: The label `owner` is required. rule require-labels failed
    at path /metadata/labels/owner/'
  policy: require-owner-labels
  result: fail
  rule: require-labels
  scored: true
  severity: medium
  source: kyverno
  timestamp:
    nanos: 0
    seconds: 1734622805`
It is PolicyReport.results.policy not PolicyReport.results.rule

Reply:
Ok, so I'm leaning towards treating it as a bug and return the correct data.
Could you create a bug in jira and separate this change to a new PR? I'm sorry for the extra work, but it can help if anyone is affected by this change.

Reply:
I created the issue. https://issues.redhat.com/browse/ACM-16775 Could you review this

Reply:
thank you!

Reply:
@jlpadilla Is this good to merge?

Reply:
Almost ready, just want to resolve this last comment. https://github.com/stolostron/search-collector/pull/349#discussion_r1894082438